background.js

```diff
@@ -68,134 +71,168 @@ function clean_amazon(url) {
-    console.debug('Entering remove_searchparams -- got url: ' + url);
-    if (url.search.length > 0) {
-        url.search = "";
-        // console.debug("Clean url to:", url.href);
-        return {redirectUrl: url.href};
-    }
-    return {redirectUrl: ''};
-}
+var remove_alisearchparams = build_query_param_remover(function (p) { return true });
 
 browser.webRequest.onBeforeRequest.addListener(
-    remove_searchparams,
+    remove_alisearchparams,
     {
         urls: [
             "*://*.aliexpress.com/item/*.html*",
             "*://*.aliexpress.com/store/product/*.html*",
+        ],
+        types: ["main_frame"]
+    }, ["blocking"]
+);
+
+var remove_fbcontentparam = build_query_param_remover(function (p) { return p == "efg" });
+browser.webRequest.onBeforeRequest.addListener(
+    remove_fbcontentparam,
+    {
+        urls: [
+            "*://*.fbcdn.net/*",
         ], types: ["main_frame"]
     }, ["blocking"]
 );
 
-function build_redirect_to_query_param(query_param_name){
-    // console.debug('Build redirect with params: ' + query_param_name);
-    const redirect_to_get_param = function(requestDetails){
+function build_redirect_to_query_param(query_param_name) {
+    const redirect_to_get_param = function (requestDetails) {
         const search_params = new URLSearchParams(new URL(requestDetails.url).search);
         const real_url_from_param = search_params.get(query_param_name);
-        if (real_url_from_param){
-            // console.debug('Redirecting to ' + real_url_from_param);
-            return {redirectUrl: real_url_from_param};
+        if (real_url_from_param) {
+            //console.log('Redirecting to ' + real_url_from_param);
+            return { redirectUrl: real_url_from_param };
         }
-        return {redirectUrl: ''};
-    };
+        return { redirectUrl: '' };
+    }
     return redirect_to_get_param;
 }
 
 const urls_to_param_mappers = [
     {
-        urls: ["*://l.facebook.com/*"],
+        urls: ["*://l.facebook.com/*", "*://lm.facebook.com/*"],
         param_name: 'u'
     },
     {
         urls: ["*://out.reddit.com/*"]
     },
     {
         urls: ["*://steamcommunity.com/linkfilter/*"]
-    }
+    },
+    {
+        urls: ["*://l.instagram.com/*"],
+        param_name: 'u'
+    },
+    {
+        urls: ["*://t.umblr.com/*"],
+        param_name: 'z'
+    },
+    {
+        urls: ["*://sys.4chan.org/derefer?*"]
+    },
+    {
+        urls: ["*://www.youtube.com/redirect?*"],
+        param_name: 'q'
+    },
 ];
 
+// Google's outbound redirect is weird so it has its own function here
+function bypass_google_redirect(requestDetails) {
+  const search_params = new URLSearchParams(new URL(requestDetails.url).search);
+  var real_url_from_param = search_params.get("q") ? search_params.get("url") : null;
+  if (real_url_from_param) {
+    //console.log('Redirecting to ' + real_url_from_param);
+    return { redirectUrl: real_url_from_param };
```

Comment:
Do you use Pocket on Firefox? They have the same mechanism to redirect using their own parser, example:

`https://getpocket.com/redirect?url=https%3A%2F%2Fwww.dw.com%2Fde%2Fcdu-vorsitz-merkel-geht-wer-kommt%2Fl-46175224&formCheck=xxxyyy`

Would it make sense to make your URL rewriter a bit more generic? I'm sure more use cases will happen for this (~for example URLs copied from google hangouts~ already managed by this)

Reply:
looking closely, you're already managing the `url` case I submitted. Do you know of any other keyword used to perform a redirect?

Reply:
This is a special one for Google. I vaguely remember finding that in the event of `?q=x&url=y`, Google will pick the `q` parameter and ignore the `url` one..

The generic redirect bypasser is already made, `build_redirect_to_query_param()`.

Reply:
To be clear, Google uses `?url=` too. So that's why we have to account for both.